control

```diff
@@ -153,64 +153,59 @@ class ControlINI(ConfigParser.SafeConfigParser, Singleton):
+
     # Default location where write() writes to if no file given
     write_path = job.resultdir
 
     # Token that signals not to execute tests
     NOEXECTOK = '!!!'
 
     def __init__(self):
-        # Help catch missing options in defaults but not in OPT_SEC_MAP
-        # by making them None instead of ''
-        super(ControlINI, self).__init__(allow_no_value=True)
+        # Inject defaults dict into ancestor's initialization
+        super(ControlINI, self).__init__(self.defaults(), allow_no_value=True)
         self.optionxform = str  # support case-sensitive options
-        for option, section in self.OPT_SEC_MAP.iteritems():
-            try:
-                self.add_section(section)
-            except ConfigParser.DuplicateSectionError:
-                pass  # already existing section
-            # Empty string is the default value
-            self.set(section, option, '')
-        # These option values default to the option name, look up section
-        for key in ('pretests', 'subtests', 'intratests', 'posttests'):
-            self.set(self.OPT_SEC_MAP[key], key, key)
+
+    def defaults(self):  # Part of the ConfigParser interface
+        """Return a dictionary containing the instance-wide defaults."""
+        return dict(include='', exclude='', subthings='',
+                    pretests='pretests', subtests='subtests',
+                    intratests='intratests', posttests='posttests',
+                    url='', username='', password='',
+                    excluded='', key_field='',
+                    key_match='', product='', component='',
+                    status='')
 
     def read(self, filenames=None):
         """
-        Read CONTROL_INI_DEFAULT or CONTROL_INI_CUSTOM if filenames is None
-        """
-        if filenames is None:
-            control_ini_default = os.path.join(self.control_path,
-                                               self.CONTROL_INI_DEFAULT)
-            control_ini_custom = os.path.join(self.control_path,
-                                              self.CONTROL_INI_CUSTOM)
-            filenames = [control_ini_default, control_ini_custom]
-        result = super(ControlINI, self).read(filenames)
-        if len(result) == 0:
-            result = ["<ControlINI defaults>"]
-        logging.debug("Loaded control configuration from %s", str(result[-1]))
+        Read default and/or custom control.init plus filesnames
+
+        :param filenames: List of filenames to attempt to load (in order)
+        returns: List of items loaded representations
+        """
+        if filenames is None:  # Preserve ancestor interface just in case
+            filenames = []
+        filenames.append(self.control_ini_custom)
+        result = ["<Empty Defaults>"]
+        # Try load defaults via readfp as documented
+        try:
+            with open(self.control_ini_default, 'rb') as defaults_file:
+                super(ControlINI, self).read(defaults_file)
```

Comment:
Sorry, this isn't an actual review; I'm still focusing on deferred deletion / mount spaces. But at a quick glance: is this correct? Running read in void context? Shouldn't this be what's added to `result`, instead of a filename?